### PR TITLE
fix(cli): format appURL from server settings (#5333)

### PR DESCRIPTION
**Pull request by yujunz**
Checklist:

* [x] This is a bug fix
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by yujunz** [comment]
The failure in e2e test does not look related to the change. Is it just flake?

https://github.com/argoproj/argo-cd/runs/1853161441?check_suite_focus=true#step:14:3451

File: cmd/argocd/commands/app.go [hunk 1]
```diff
@@ -361,18 +361,12 @@ func printAppConditions(w io.Writer, app *argoappv1.Application) {
...
-			server = server[0 : len(server)-4]
-		}
-	}
-	return fmt.Sprintf("%s://%s/applications/%s", scheme, server, appName)
+	conn, settingsIf := acdClient.NewSettingsClientOrDie()
+	defer argoio.Close(conn)
+	argoSettings, err := settingsIf.Get(context.Background(), &settingspkg.SettingsQuery{})
+	errors.CheckError(err)
+
+	return fmt.Sprintf("%s/applications/%s", argoSettings.URL, appName)
```
Comment (alexmt) [inline cmd/argocd/commands/app.go:369]: The URL might be empty in settings. Please use URL resolved using `opts.ServerAddr` as a fallback.

**Review by alexmt — CHANGES_REQUESTED** [review]

File: cmd/argocd/commands/app.go (hunk 1 above)
Comment (yujunz) [reply cmd/argocd/commands/app.go]: Fixed.

**Review by alexmt — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**